src/test/firebase-resource-manager.spec.ts

```diff
@@ -0,0 +1,420 @@
+import { expect } from "chai";
+import * as sinon from "sinon";
+
+import * as api from "../api";
+import {
+  FirebaseResourceManager,
+  ParentResource,
+  ParentResourceType,
+} from "../firebase-resource-manager";
+import { mockAuth } from "./helpers";
+import { OraWrapper } from "../oraWrapper";
+
+const PROJECT_ID = "the-best-firebase-project";
+const PROJECT_NUMBER = "1234567890";
+const PROJECT_NAME = "The Best Project";
+const PARENT_RESOURCE: ParentResource = { id: "1111111111111", type: ParentResourceType.FOLDER };
+
+// TODO(caot): Removed when "Deferred Analytics" and "Deferred Location" are launched
+const TIME_ZONE = "America/Los_Angeles";
+const REGION_CODE = "US";
+const LOCATION_ID = "us-central";
+
+describe("FirebaseResourceManager", () => {
+  describe("createFirebaseProject", () => {
+    let sandbox: sinon.SinonSandbox;
+    let mockApi: sinon.SinonMock;
+    let mockPoller: sinon.SinonMock;
+    let mockOraWrapper: sinon.SinonMock;
+    let expectedResolvedValues = [];
+    const firebaseResourceManager = new FirebaseResourceManager();
+
+    beforeEach(() => {
+      sandbox = sinon.createSandbox();
+      mockAuth(sandbox);
+      mockApi = sandbox.mock(api);
+      mockOraWrapper = sandbox.mock(OraWrapper.prototype);
+      mockPoller = sandbox.mock((firebaseResourceManager as any).poller);
+    });
+
+    afterEach(() => {
+      mockApi.verify();
+      mockPoller.verify();
+      mockOraWrapper.verify();
+      sandbox.restore();
+      expectedResolvedValues = [];
+    });
+
+    it("should resolve with project data if it succeeds", async () => {
+      expectedResolvedValues = [
+        { body: { name: "operations/cp.111111111111111" } },
+        { response: { projectNumber: PROJECT_NUMBER } },
+        { body: { name: "operations/cp.222222222222223" } },
+        { response: { projectId: PROJECT_ID, displayName: PROJECT_NAME } },
+      ];
+      mockApi
+        .expects("request")
+        .withArgs("POST", "/v1/projects", {
+          auth: true,
+          origin: api.resourceManagerOrigin,
+          timeout: 15000,
+          data: { projectId: PROJECT_ID, name: PROJECT_NAME, parent: PARENT_RESOURCE },
+        })
+        .once()
+        .resolves(expectedResolvedValues[0]);
+      mockApi
+        .expects("request")
+        .withArgs("POST", `/v1beta1/projects/${PROJECT_ID}:addFirebase`, {
+          auth: true,
+          origin: api.firebaseApiOrigin,
+          timeout: 15000,
+          data: { timeZone: TIME_ZONE, regionCode: REGION_CODE, locationId: LOCATION_ID },
+        })
+        .once()
+        .resolves(expectedResolvedValues[2]);
+      mockPoller
+        .expects("poll")
+        .withArgs({
+          pollerName: "Project Creation Poller",
+          apiOrigin: api.resourceManagerOrigin,
+          apiVersion: "v1",
+          operationResourceName: expectedResolvedValues[0].body!.name,
```

Comment:
I would prefer to get rid of the `expectedResolvedValues` array, and instead create constants for values like `"operations/cp.111111111111111"` which are referenced inline when needed.

Reply:
Done